docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,203 @@ Note that:
+         
+  * external device child1: Uploads configuration file to the thin-edge device with the HTTP filetransfer feature.
+  
+    HTTP PUT request: `http://<ip address of thin-edge devicee>/tedge/tmpfiles`
+    
+    The response from the HTTP filetransfer feature contains a _temporary URL_, that is later used to uniquely access that uploaded file.<br/>
+    Example for a _temporary URL_: `/tedge/tmpfiles/<random string>`
+
+  * external device child1: notifies the the plugin via MQTT about succeeded upload and the _temporary URL_.
+ 
+    Topic: `tedge/configuration/res/retrieve/{config type}/{childid}`<br/>
+    Example: `tedge/configuration/res/retrieve/bar.conf/child1`<br/>
+    Payload: ` <temporary URL> `
+       
+    TODO: Investigate and decide about topic structure and payload in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307). Unhappy paths also to be considered here.
+  
+  * C8Y config plugin: recognizes the MQTT notification about the uploaded file and the temprary-URL, 
+                       and uses the filetransfers local API to obtain the file from the _temprary URL_ to some plugin specific local filesystem path (e.g. `/tmp/c8y-cfg-plugin/<childid>_<cfg-type>`.
+
+    * TODO: Investigate and decide about filetransfer's local API in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307), according to topics below:
+ 
+      * local API must avoid that there are two copies of the same file at any point in time
+      * local API must avoid that the file content is transfered/copied from one to another place, when the plugin _obtains_ the file. 
+        Instead just the directory link may change (as it is the case for bash `mv` command).
+      * file is removed from filetransfer's specific folder when the file was successfully obtained to the plugin's path.
+      
+      Reasons for requirements above: Saving disc-space and CPU resources for large files.
+
+    * TO-BE-DEFINED:
+    
+      * Decide which error paths to be considered.
+      * Decide if any timeout to be considerd when waiting for upload notification from external device.
+
+  * C8Y config plugin: sends the obtained file to C8Y
+
+  * C8Y config plugin: removed the file form local filesystem
+  
+
+**Cloud-to-Device Behaviour:**
+  * at some point a config sent from cloud for type `bar.conf` for `child1` arrives at C8Y config plugin<br/>
+    Format of C8Y SmartREST message for config send operation: `524,<childid>,<URL>,<config type>`. See [C8Y SmartREST doc](https://cumulocity.com/guides/reference/smartrest-two/#download-configuration-file-with-type-524)<br/>
+    Example: `524,child1,http://www.my.url,bar.conf`
+  * C8Y config plugin: downloads the file based on the URL received from C8Y, and stores it some plugin specific local filesystem path (e.g. `/tmp/c8y-cfg-plugin/<childid>_<cfg-type>`).
```

Comment:
```suggestion
  * C8Y config plugin: downloads the file based on the URL received from C8Y, to some temporary location
```
Removing the example path as well, since the location or the name of the temporary file doesn't matter and hence we don't need to enforce a naming convention.